src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -115,6 +135,11 @@ private void QuickFail(Transaction tx, BlockHeader block, ITxTracer txTracer, bo
                 
                 txTracer.MarkAsFailed(recipient, tx.GasLimit, Array.Empty<byte>(), reason ?? "invalid", stateRoot);
             }
+
+            if (deleteCallerAccount)
+            {
+                _stateProvider.DeleteAccount(tx.SenderAddress ?? throw new InvalidOperationException());
+            }
```

Comment:
Maybe this should be put in a finally clause?

Reply:
```
        private void QuickFail(Transaction tx, BlockHeader block, ITxTracer txTracer, bool eip658NotEnabled, string? reason, bool deleteCallerAccount = false)
        {
            try
            {
                block.GasUsed += tx.GasLimit;
            
                Address recipient = tx.To ?? ContractAddress.From(
                    tx.SenderAddress ?? Address.Zero,
                    _stateProvider.GetNonce(tx.SenderAddress ?? Address.Zero));
            
                if (txTracer.IsTracingReceipt)
                {
                    Keccak? stateRoot = null;
                    if (eip658NotEnabled)
                    {
                        _stateProvider.RecalculateStateRoot();
                        stateRoot = _stateProvider.StateRoot;
                    }
                
                    txTracer.MarkAsFailed(recipient, tx.GasLimit, Array.Empty<byte>(), reason ?? "invalid", stateRoot);
                }

            }
            finally
            {
                if (deleteCallerAccount)
                {
                    _stateProvider.DeleteAccount(tx.SenderAddress ?? throw new InvalidOperationException());
                }
            }
        }

```
Do something like this?

Reply:
yes

Reply:
ok